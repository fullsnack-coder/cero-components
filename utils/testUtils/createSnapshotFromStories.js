import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
