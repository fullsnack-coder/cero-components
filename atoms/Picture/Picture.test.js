import * as stories from './Picture.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Picture ]', () => {
  describe('#Picture', () => createSnapshotFromStories('Picture', stories))
})
