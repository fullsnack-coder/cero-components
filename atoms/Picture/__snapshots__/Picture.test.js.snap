// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Picture ] #Picture when 'Picture' Border is mounted should render 1`] = `
<DocumentFragment>
  <picture
    class="picture with-border"
  >
    <img
      src="https://picsum.photos/id/1033/400/400"
      style="height: auto; max-width: 200px;"
    />
  </picture>
</DocumentFragment>
`;

exports[`[ atoms / Picture ] #Picture when 'Picture' Default is mounted should render 1`] = `
<DocumentFragment>
  <picture
    class="picture"
  >
    <img
      src="https://picsum.photos/id/1033/400/400"
      style="height: auto; max-width: 200px;"
    />
  </picture>
</DocumentFragment>
`;

exports[`[ atoms / Picture ] #Picture when 'Picture' Height is mounted should render 1`] = `
<DocumentFragment>
  <picture
    class="picture"
  >
    <img
      src="https://picsum.photos/id/1033/400/800"
      style="height: 400px; max-width: 200px;"
    />
  </picture>
</DocumentFragment>
`;

exports[`[ atoms / Picture ] #Picture when 'Picture' Rounded is mounted should render 1`] = `
<DocumentFragment>
  <picture
    class="picture is-rounded"
  >
    <img
      src="https://picsum.photos/id/1033/400/400"
      style="height: auto; max-width: 200px;"
    />
  </picture>
</DocumentFragment>
`;

exports[`[ atoms / Picture ] #Picture when 'Picture' Width is mounted should render 1`] = `
<DocumentFragment>
  <picture
    class="picture"
  >
    <img
      src="https://picsum.photos/id/1033/800/400"
      style="height: auto; max-width: 400px;"
    />
  </picture>
</DocumentFragment>
`;
